.github/workflows/check.yaml

```diff
@@ -147,4 +147,25 @@ jobs:
         with:
           name: test-results
           path: testresults--all-features.json
-          
\ No newline at end of file
+
+  doctest:
+    # Run doctests separately, as nextest doesn't yet (https://github.com/nextest-rs/nextest/issues/16)
+    outputs:
+      test_results_url: ${{ steps.doctest_results.outputs.artifact-url }}
+    runs-on: ubuntu-latest
+    env:
+      RUSTDOCFLAGS: -Dwarnings
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@master
+        with: 
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+      - name: Run doc tests
+        run: |
+          cargo test --doc --all-features
```

Comment:
if we create JSON output from nextest, maybe we also want to create JSON output here?